### ISSUE TITLE
I cannot run cypress-lambda-example in AWS lambda

### ISSUE THREAD

**Issue by manv6** [magJ/cypress-lambda-example#1]
@magJ I deployed your code to aws and when testing it from aws console I'm getting 

```
START RequestId: 76af2da6-e8fe-4e79-a6f3-8ba680cfc46b Version: $LATEST
2022-10-21T06:51:38.641Z	76af2da6-e8fe-4e79-a6f3-8ba680cfc46b	INFO	Running cypress tests with electron
prctl(PR_SET_NO_NEW_PRIVS) failed
prctl(PR_SET_NO_NEW_PRIVS) failed
[24:1021/065142.756131:ERROR:gpu_process_host.cc(969)] GPU process launch failed: error_code=1002
[24:1021/065142.760996:ERROR:gpu_process_host.cc(969)] GPU process launch failed: error_code=1002
[24:1021/065142.762894:ERROR:gpu_process_host.cc(969)] GPU process launch failed: error_code=1002
[24:1021/065144.429972:ERROR:gpu_process_host.cc(969)] GPU process launch failed: error_code=1002
[24:1021/065144.498503:ERROR:gpu_process_host.cc(969)] GPU process launch failed: error_code=1002
[24:1021/065144.500677:ERROR:gpu_process_host.cc(969)] GPU process launch failed: error_code=1002
[24:1021/065144.500712:FATAL:gpu_data_manager_impl_private.cc(441)] GPU process isn't usable. Goodbye.
2022-10-21T06:51:47.190Z	76af2da6-e8fe-4e79-a6f3-8ba680cfc46b	ERROR	Invoke Error 	{"errorType":"Error","errorMessage":"The Test Runner unexpectedly exited via a close event with signal SIGILL\n\nPlease search Cypress documentation for possible solutions:\n\nhttps://on.cypress.io\n\nCheck if there is a GitHub issue describing this crash:\n\nhttps://github.com/cypress-io/cypress/issues\n\nConsider opening a new issue.\n\n----------\n\nPlatform: linux-x64 (Amazon Linux - 2 (Karoo))\nCypress Version: 10.10.0","known":true,"stack":["Error: The Test Runner unexpectedly exited via a close event with signal SIGILL","","Please search Cypress documentation for possible solutions:","","https://on.cypress.io","","Check if there is a GitHub issue describing this crash:","","https://github.com/cypress-io/cypress/issues","","Consider opening a new issue.","","----------","","Platform: linux-x64 (Amazon Linux - 2 (Karoo))","Cypress Version: 10.10.0","    at /var/task/node_modules/cypress/lib/errors.js:277:17"]}
END RequestId: 76af2da6-e8fe-4e79-a6f3-8ba680cfc46b
REPORT RequestId: 76af2da6-e8fe-4e79-a6f3-8ba680cfc46b	Duration: 8562.25 ms	Billed Duration: 10542 ms	Memory Size: 3008 MB	Max Memory Used: 498 MB	Init Duration: 1978.97 ms	
XRAY TraceId: 1-63524178-3a19c3ef7aed7fc435f76946	SegmentId: 604dbdd1219ef6d3	Sampled: true	
```
and this is the docker build output 
```
Step 1/17 : FROM public.ecr.aws/lambda/nodejs:16
16: Pulling from lambda/nodejs
Digest: sha256:5e044c7b1a3efa29411dcc02ff1a9f2ff6c967d4fe26b79020af1c8a12a40ccd
Status: Image is up to date for public.ecr.aws/lambda/nodejs:16
 ---> 017b9d135ea2
Step 2/17 : RUN yum install -y     xorg-x11-server-Xvfb     gtk2-devel     gtk3-devel     libnotify-devel     GConf2     nss     libXScrnSaver     alsa-lib     binutils &&     yum clean all
 ---> Using cache
 ---> bf856f12a452
Step 3/17 : RUN yum install -y     amazon-linux-extras &&     amazon-linux-extras install epel -y &&     yum install -y     chromium &&     yum clean all
 ---> Using cache
 ---> a5274377f65f
Step 4/17 : ENV CYPRESS_CACHE_FOLDER=/opt/cypress
 ---> Using cache
 ---> 2d278454a03c
Step 5/17 : ENV XDG_CONFIG_HOME=/tmp
 ---> Using cache
 ---> d58afe4601eb
Step 6/17 : RUN mkdir $CYPRESS_CACHE_FOLDER
 ---> Using cache
 ---> 1584a41ed6bc
Step 7/17 : COPY lambda2/package.json lambda2/package-lock.json ./
 ---> b131bd7a21cc
Step 8/17 : COPY lambda2/patch-cypress.sh .
 ---> b46f8f9839d4
Step 9/17 : RUN npm i && chmod -R +r $CYPRESS_CACHE_FOLDER && ./patch-cypress.sh
 ---> Running in c72976479242

added 167 packages, and audited 168 packages in 23s

28 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice 
npm notice New minor version of npm available! 8.15.0 -> 8.19.2
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.19.2>
npm notice Run `npm install -g npm@8.19.2` to update!
npm notice 
9+0 records in
9+0 records out
9 bytes (9 B) copied, 5.2099e-05 s, 173 kB/s
9+0 records in
9+0 records out
9 bytes (9 B) copied, 4.7398e-05 s, 190 kB/s
9+0 records in
9+0 records out
9 bytes (9 B) copied, 4.7827e-05 s, 188 kB/s
9+0 records in
9+0 records out
9 bytes (9 B) copied, 4.5238e-05 s, 199 kB/s
9+0 records in
9+0 records out
9 bytes (9 B) copied, 4.1075e-05 s, 219 kB/s
Removing intermediate container c72976479242
 ---> 9edc5d922e15
Step 10/17 : COPY lambda2/cypress.config.ts lambda2/tsconfig.json ./
 ---> 4ef290469c6c
Step 11/17 : COPY lambda2/tests ./tests
 ---> da3abd906366
Step 12/17 : COPY lambda2/src ./src
 ---> bb4710f0276c
Step 13/17 : RUN npx cypress verify
 ---> Running in eb2bf3e981d9

[STARTED] Task without title.
[TITLE]  Verified Cypress!       /opt/cypress/10.10.0/Cypress
[SUCCESS]  Verified Cypress!       /opt/cypress/10.10.0/Cypress
Removing intermediate container eb2bf3e981d9
 ---> e22121421e22
Step 14/17 : RUN npx tsc
 ---> Running in 14e9a05d6df5
Removing intermediate container 14e9a05d6df5
 ---> 3a68f840a0bc
Step 15/17 : RUN rm cypress.config.js
 ---> Running in 5499fe32acc2
Removing intermediate container 5499fe32acc2
 ---> 676248e33a09
Step 16/17 : RUN /sbin/adduser testuser
 ---> Running in e72a8947ccc1
Removing intermediate container e72a8947ccc1
 ---> 942d2e3b8f08
Step 17/17 : CMD ["src/lambdaHandler.handler"]
 ---> Running in b0a3d40c045d
Removing intermediate container b0a3d40c045d
 ---> 9cb1693b193a
Successfully built 9cb1693b193a
Successfully tagged 437795906767.dkr.ecr.us-east-1.amazonaws.com/lambda-cypress-tests-containerized:db574ad2e99bd1d5771d3232d3e6b4833e46b459
```

Am I missing something here?

**Comment by magJ** [comment]
Sorry, I might have missed something in the more recent commits.  

I’ve only verified the latest changes on a private project setup in a similar way, but might have missed something porting it to this repo.

I verified that the solution in the initial commit works as-is(the majority of the time, although I found that it still occasionally fails initialising electron).  

Looking at your error message, you might need a disable GPU flag.  

I’m out at the moment, but when I get back I’ll verify the current master end to end, and update anything I missed.

**Comment by magJ** [comment]
@manv6 My claims about getting electron to work from lambda were wrong, and based on flawed and incomplete local testing.  
I've gone and updated the code, and any comments I've made elsewhere that incorrectly asserted that to be the case.  

I've updated the code to only run tests using chromium.  

I've kept the electron patching logic, even though in my original implementation I found that it works without it.
The reason I've kept it, is that I've noticed that occasionally(maybe 1 in 50 runs) electron will still fail to start, my hope is that patching `/dev/shm` might address that, although I'm not sure.

I'm still interested in getting electron to work fully, and if I can figure that out, I'll update things accordingly.
